app/controllers/projects/blob_controller.rb

```diff
@@ -146,10 +146,19 @@ def editor_variables
 
     @file_path =
       if action_name.to_s == 'create'
+        if params[:file_upload].present?
+          params[:file_name] = params[:file_upload].original_filename
+        end
         File.join(@path, File.basename(params[:file_name]))
       else
         @path
       end
+     
+    if params[:file_upload].present?
+      params[:content] = params[:file_upload].read
+      params[:encoding] = 'text'
```

Comment:
I just tested this out. If you replace an image with another image, the replaced file is corrupted.